### PR TITLE
SN-API-42 Add relationship

### PR DESCRIPTION
**Pull request by Aretemal**

### CODE REVIEW

File: src/serializers/CollectionSerializer.js [hunk 1]
```diff
@@ -25,13 +17,35 @@ class CollectionSerializer extends Serializer {
...
-  links() {
-    return { self: `${process.env.API_URL}/${this.request.originalUrl}` };
+  included() {
+    const includeArray = [];
+    this.include.forEach((includedName) => {
+      this.resource.forEach((serializable) => {
+        const IncludedClass = this.getter(includedName);
+        const serializer = new IncludedClass(serializable[includedName], { include: this.include });
+        const { data } = serializer.serialize();
+        if (!(includeArray.some((item) => item.id === data.id))) {
```
Comment (Konstantsin) [inline src/serializers/CollectionSerializer.js:35]: What is the comnplexity due to this operation in loop? Can we have other data structure to be more performant here?

File: src/serializers/CollectionSerializer.js [hunk 2]
```diff
@@ -25,13 +17,35 @@ class CollectionSerializer extends Serializer {
...
     });
   }
 
-  links() {
-    return { self: `${process.env.API_URL}/${this.request.originalUrl}` };
+  included() {
+    const includeArray = [];
+    this.include.forEach((includedName) => {
+      this.resource.forEach((serializable) => {
+        const IncludedClass = this.getter(includedName);
```
Comment (Konstantsin) [inline src/serializers/CollectionSerializer.js:32]: Can we move it out of loop?

File: src/serializers/MessageSerializer.js [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
+    this.include.forEach((relationName) => {
+      const RelationClass = this.getter(relationName);
+      const serializer = new RelationClass(this.resource[relationName], { include: this.include });
+      data[relationName] = {
+        links: serializer.serialize().data.links,
+        data: [{
+          id: serializer.serialize().data.id,
+          type: serializer.serialize().data.type,
+        }],
+      };
```
Comment (Konstantsin) [inline src/serializers/MessageSerializer.js:35]: can we have next structure here ? 
```
serializer = new RelationClass(this.resource[relationName], { include: this.include, only:['id', 'type']  })
```

where serializer will only render `id` and `type` in this case.

File: src/serializers/MessageSerializer.js [hunk 4]
```diff
@@ -1,14 +1,46 @@
...
+    if (this.include) {
+      data.data.relationships = this.relationships();
+    }
+    return data;
+  }
+
+  relationships() {
+    const data = {};
+    this.include.forEach((relationName) => {
+      const RelationClass = this.getter(relationName);
```
Comment (Konstantsin) [inline src/serializers/MessageSerializer.js:27]: Can we move it out of the loop?

File: src/serializers/CollectionSerializer.js (hunk 2 above)
Comment (Aretemal) [reply src/serializers/CollectionSerializer.js:32]: No

File: src/serializers/CollectionSerializer.js (hunk 1 above)
Comment (Aretemal) [reply src/serializers/CollectionSerializer.js:35]: done

File: src/serializers/CollectionSerializer.js (hunk 2 above)
Comment (Aretemal) [reply src/serializers/CollectionSerializer.js:32]: done

File: src/serializers/MessageSerializer.js (hunk 4 above)
Comment (Aretemal) [reply src/serializers/MessageSerializer.js:27]: No. This needs data from the loop.

File: src/serializers/MessageSerializer.js (hunk 3 above)
Comment (Aretemal) [reply src/serializers/MessageSerializer.js:35]: I don't quite understand how this is supposed to work.

**Pull request merged**